controller/backup_controller.go

```diff
@@ -385,146 +381,185 @@ func (bc *BackupController) getBackupVolumeName(backup *longhorn.Backup) (string
+			biName, bv.Status.BackingImageChecksum, bi.Status.Checksum)
+	}
+	return bi.Status.Checksum, nil
+}
+
+func (bc *BackupController) backupCreation(backupTarget *longhorn.BackupTarget, volume *longhorn.Volume, backup *longhorn.Backup) (skip bool, err error) {
+	log := bc.logger.WithFields(
 		logrus.Fields{
-			"vol":      volumeName,
+			"backup":   backup.Name,
+			"vol":      volume.Name,
 			"snapshot": backup.Spec.SnapshotName,
 			"label":    backup.Spec.Labels,
 		},
 	)
 
-	event := func(err error, state longhorn.BackupState, backup *longhorn.Backup, volume *longhorn.Volume) {
+	currentState := backup.Status.State
+	defer func() {
+		if currentState == backup.Status.State {
+			return
+		}
 		if err != nil {
-			bc.eventRecorder.Eventf(volume, corev1.EventTypeWarning, string(state),
+			bc.eventRecorder.Eventf(volume, corev1.EventTypeWarning, string(backup.Status.State),
 				"Snapshot %s backup %s label %v: %v", backup.Spec.SnapshotName, backup.Name, backup.Spec.Labels, err)
 			return
 		}
-		bc.eventRecorder.Eventf(volume, corev1.EventTypeNormal, string(state),
+		bc.eventRecorder.Eventf(volume, corev1.EventTypeNormal, string(backup.Status.State),
 			"Snapshot %s backup %s label %v", backup.Spec.SnapshotName, backup.Name, backup.Spec.Labels)
-	}
-
-	// Get the volume CR
-	volume, err := bc.ds.GetVolume(volumeName)
-	if err != nil {
-		return err
-	}
+	}()
 
-	// Backing image validation
-	biName := volume.Spec.BackingImage
-	biChecksum := ""
-	if biName != "" {
-		bi, err := bc.ds.GetBackingImage(biName)
+	switch backup.Status.State {
+	case longhorn.BackupStateNew:
+		// Backing image checksum validation
+		biChecksum, err := bc.validateBackingImageChecksum(volume.Name, volume.Spec.BackingImage)
 		if err != nil {
-			return err
-		}
-		bv, err := bc.ds.GetBackupVolumeRO(volumeName)
-		if err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-		if bv != nil &&
-			bv.Status.BackingImageChecksum != "" && bi.Status.Checksum != "" &&
-			bv.Status.BackingImageChecksum != bi.Status.Checksum {
-			return fmt.Errorf("the backing image %v checksum %v in the backup volume doesn't match the current checksum %v", biName, bv.Status.BackingImageChecksum, bi.Status.Checksum)
+			return true, err
 		}
-		biChecksum = bi.Status.Checksum
-	}
-
-	backup.Status.State = longhorn.BackupStateInProgress
-	event(nil, backup.Status.State, backup, volume)
 
-	go func() {
-		state := backup.Status.State
-		defer func() {
-			backup, err := bc.ds.GetBackup(backup.Name)
-			if err != nil {
-				log.WithError(err).Errorf("Error get backup")
-				return
-			}
-			existingBackup := backup.DeepCopy()
+		// Initialize a backup target client
+		backupTargetClient, err := getBackupTargetClient(bc.ds, backupTarget)
+		if err != nil {
+			log.WithError(err).Error("Error init backup target client")
+			return true, nil // Ignore error to prevent enqueue
```

Comment:
Why the exact reason is for ignoring the error from `getBackupTargetClient`?

Reply:
Because the function `getBackupTargetClient` returns error when the URL+Credential pair in invalid.
Skip return error here to prevent the controller from re-enqueue it again because even after immediately reconciling again, the backup target client is still invalid.

Reply:
Got it. Maybe it will be good to check the exact error type which you expect to be able to ignore?

Reply:
Fair.

I'd prefer to create another PR for this one because the backup_target_controller/backup_volume_controller/backup_controller all use the same function `getBackupTargetClient `.

Reply:
Sure, please do it.